src/gtl/interfaces/abstractDataViewClass.ts

```diff
@@ -81,6 +81,17 @@ export abstract class DataViewClass implements IDataTableBinding {
       `${start} - ${end} of ${total}`;
   }
 
+  /**
+   * Helper method to count all checkboxes in rows data.
+   * Checkboxes are stored under each row's cells.
+   */
+  public countCheckboxes() {
+    return this.rows.reduce(
+      (curr: number, next) => next.cells && next.cells.filter(oneCell => oneCell && oneCell.is_checkbox).length + curr,
```

Comment:
Should this be `curr + (next.cells && ...)`?

Because, now any row with no cells will reset the counter to `undefined` or `NaN`.

Reply:
Good catch. However this is even better and more robust
```javascript
return this.rows.reduce(
  (curr: number, next) => {
    if (next.cells) {
      curr += next.cells.filter(oneCell => oneCell && oneCell.is_checkbox).length
    }
    return curr;
  },
  0
);
```

Reply:
And tests written to cower this isse as well.